pioinstaller/core.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2014-present PlatformIO <contact@platformio.org>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from pioinstaller import util
+
+
+def _get_core_dir():
+    if os.getenv("PLATFORMIO_CORE_DIR"):
+        return os.getenv("PLATFORMIO_CORE_DIR")
+
+    core_dir = os.path.join(util.expanduser("~"), ".platformio")
+    if not util.IS_WINDOWS or not util.has_non_ascii_char(core_dir):
+        return core_dir
+
+    core_dir = os.path.splitdrive(core_dir)[0] + "\\.platformio"
+
+    if not os.path.isdir(core_dir):
+        os.makedirs(core_dir)
+        with open(os.path.join(core_dir, "test"), "w") as fp:
+            fp.write("test")
```

Comment:
Move under try ecept/bare and return `core_dir` on exception. If OK makedirs `win_core_dir`, remove `file.tmp`

rename `test` to `file.tmp`